lib/chef/resource/windows_pagefile.rb

```diff
@@ -137,21 +152,27 @@ def exists?(pagefile)
         # @param [String] max the minimum size of the pagefile
         # @return [Boolean]
         def max_and_min_set?(pagefile, min, max)
-          @max_and_min_set ||= begin
-            logger.trace("Checking if #{pagefile} min: #{min} and max #{max} are set")
-            cmd = shell_out("wmic.exe pagefileset where SettingID=\"#{get_setting_id(pagefile)}\" list /format:list", returns: [0])
-            cmd.stderr.empty? && (cmd.stdout =~ /InitialSize=#{min}/i) && (cmd.stdout =~ /MaximumSize=#{max}/i)
-          end
+          logger.trace("Checking if #{pagefile} has max and initial disk size values set")
+          cmd =  "$page_file = '#{pagefile}';"
+          cmd << "$driveLetter = $page_file.split(':')[0];"
+          cmd << "$page_file_settings = Get-CimInstance -ClassName Win32_PageFileSetting -Filter \"SettingID='pagefile.sys @ $($driveLetter):'\" -Property * -ErrorAction Stop;"
+          cmd << "if ($page_file_settings.InitialSize -eq #{min} -and $page_file_settings.MaximumSize -eq #{max})"
+          cmd << "{ return $true }"
+          cmd << "else { return $false }"
+          powershell_exec!(cmd).result
         end
 
         # create a pagefile
         #
         # @param [String] pagefile path to the pagefile
         def create(pagefile)
           converge_by("create pagefile #{pagefile}") do
-            logger.trace("Running wmic.exe pagefileset create name=\"#{pagefile}\"")
-            cmd = shell_out("wmic.exe pagefileset create name=\"#{pagefile}\"")
-            check_for_errors(cmd.stderr)
+            logger.trace("Running New-CimInstance -ClassName Win32_PageFileSetting to create new pagefile : #{pagefile}")
+            powershell_exec! <<~ELM
+              $page_file = "#{pagefile}"
+              $driveLetter = $page_file.split(':')[0]
+              New-CimInstance -ClassName Win32_PageFileSetting -Property  @{Name = "$($DriveLetter)\:\\pagefile.sys"}
+            ELM
```

Comment:
it seems weird that we are just using the drive letter and ignore the rest of the user supplied path.

Reply:
I had assumed from all the examples I read that the pagefile had to exist in the root of whatever drive you put it in.

Reply:
yeah maybe. If thats so it seems we should validate what the user provided. Does it have to be named `pagefile.sys`?

Reply:
I can't find an example of where it is NOT named pagefile.sys. If you go through the Advanced settings in the GUI and tell the OS to make you pagefile on D with a min and max setting you'll get a d:\pagefile.sys with those settings.

Reply:
then it seems like the resource should not ask for a path at all and just a drive letter. It's probably best to experiment locally with different paths and see what happens.

Reply:
ooh, good points. Will do that for sure.